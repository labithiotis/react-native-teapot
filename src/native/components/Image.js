import React, { Component, PropTypes } from 'react';
import ImageProgress from 'react-native-image-progress';
import CircleSnail from 'react-native-progress/CircleSnail';
import { colours } from '../styles';
import isAbsoluteUrl from 'is-absolute-url';

export default class Image extends Component {
  static propTypes = {
    indicator: PropTypes.element,
    indicatorProps: PropTypes.object,
    indicatorSize: PropTypes.number,
    threshold: PropTypes.number,
    onError: PropTypes.func,
    source: PropTypes.oneOfType([PropTypes.number, PropTypes.object])
      .isRequired,
    sourceFallBack: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
      .isRequired,
  };

  static defaultProps = {
    onError: () => {},
    sourceFallBack: require('./img/EmptyState@2x.png'),
  };

  constructor(props) {
    super(props);
    this.onError = this.onError.bind(this);
    const { source, sourceFallBack } = props;
    this.state = {
      source: source && source.uri && !isAbsoluteUrl(source.uri)
        ? sourceFallBack
        : source,
    };
  }

  onError(e) {
    const { onError, sourceFallBack } = this.props;
    this.setState({ source: sourceFallBack });
    onError(e);
  }

  render() {
    return (
      <ImageProgress
        {...this.props}
        source={this.state.source}
        indicator={CircleSnail}
        indicatorProps={{
          size: this.props.indicatorSize || 60,
          color: colours.blue,
        }}
        threshold={300}
        onError={this.onError}
      />
    );
  }
}
