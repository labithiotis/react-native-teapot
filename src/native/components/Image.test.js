import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

describe('Image', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Image source={require('./img/EmptyState.png')} />,
    );
    expect(component).toMatchSnapshot();
  });
});
