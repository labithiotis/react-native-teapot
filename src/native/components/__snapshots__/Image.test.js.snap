// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `
<ImageProgress
  indicator={[Function]}
  indicatorProps={
    Object {
      "color": "#3262a6",
      "size": 60,
    }
  }
  onError={[Function]}
  source={1}
  sourceFallBack={1}
  threshold={300}
/>
`;
